specification/appconfiguration/AppConfiguration/routes.tsp

```diff
@@ -36,6 +36,7 @@ alias syncTokenHeader = {
 alias contentTypeHeader<T extends string> = {
   @doc("Content-Type header")
   @header("Content-Type")
+  @friendlyName("contentType")
   contentTypeHeader?: T | "application/problem+json";
```

Comment:
Why not rename the property directly to `contentType`?

I didn't find `contentTypeHeader` in swagger, so I assume the `contentTypeHeader` does not matter.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json

Reply:
@catalinaperalta , any reasons why we couldn't just name is `contentType` ?

Reply:
I dont think there should be a problem renaming this directly. Try renaming it to just contentType and make sure to regenerate the openapi spec since this would change it (run `tsp compile .`) and push the update openapi spec to this PR since you need to keep it in sync with the tsp definitions. Let me know if you run into any issues.

Reply:
One possible cause is that there is another `contentType` in body (and that's flattened).

One caution is that the `contentTypeHeader` (after `@friendlyName`) and `contentType` (in body) would have a same name. We may want make sure JS dev is OK with it (I think JS may have named property of headers in response).

Reply:
@lmazuel @catalinaperalta Yesterday I tried to rename tsp directly but meet compiler error since there is already a property named "contentType" in "[KeyValue](https://github.com/Azure/azure-rest-api-specs/blob/c65e8cbbb18fc9d79165f12291d38d91eb340f0f/specification/appconfiguration/AppConfiguration/models.tsp#L182)"

![image](https://github.com/user-attachments/assets/95e719e9-e502-4467-81bf-8a07f1cf9e65)

Then I find the easiest way is to use `@friendlyName`.

Reply:
> One possible cause is that there is another `contentType` in (implicit) body.
> 
> One caution is that the `contentTypeHeader` (after `@friendlyName`) and `contentType` (in body) would have a same name. We may want to make sure JS dev is OK with it (I think JS may have named property of headers in response).

Add JS owner @MaryGao @qiaozha for awareness.

Reply:
Talked with @timotheeguerin offline, we can make this works without the friendly with a slight change in the operation return definition:
```typescript
  Body<KeyValue> &
    AppConfigResponseHeaders &
    contentTypeHeader<"application/vnd.microsoft.appconfig.kv+json">
```

See the addition of `Body` around `KeyVault`, which helps the TypeSpec emitter to understand there should be no conflict.

Reply:
After update with @timotheeguerin's suggestion, only one error left: 
![image](https://github.com/user-attachments/assets/65db02d6-d3ea-4520-892f-53c61fc554b8)

It is related with Template `ResourceRead`:

![image](https://github.com/user-attachments/assets/a8f0396d-d619-4663-9221-6c638a77bd74)

I understand that we need to customize this template like:?

```
  @Foundations.Private.ensureVerb("ResourceRead", "GET")
  @readsResource(Resource)
  ResourceRead<
    Resource extends TypeSpec.Reflection.Model,
    Traits extends TypeSpec.Reflection.Model = {}
  > is Foundations.ResourceOperation<
    Resource,
    TraitProperties<Traits & InterfaceTraits, TraitLocation.Parameters, TraitContext.Read>,
    Body<Resource> &
      TraitProperties<Traits & InterfaceTraits, TraitLocation.Response, TraitContext.Read>,
    Traits & InterfaceTraits,
    ErrorResponse
  >;
```

Reply:
Can you show how this template is called currently where it cause this error

Reply:
@msyyc we shouldnt use internal decorators in service specs, you can create a custom template if you need, but maybe we can just use the clientName decorator?

Reply:
Yeah should definitely not call any internal decorators those will have no non breaking requirement.

In the case of this template this decorator is just to add some validation so you would just not use it but I don't think a custom template is the right solution here.